### ISSUE TITLE
Add Right-Click Shortcut to Create Shader Material from Shader File

### ISSUE THREAD

**Issue by realkotob** [godotengine/godot-proposals#6028]
### Describe the project you are working on

N/A

### Describe the problem or limitation you are having in your project

Creating Shader Materials takes too many clicks. Assuming the Shader resource already exists, I need to create a new resource, select the Shader Material, etc.., and then I have to attach the Shader resource to it.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add right-click shortcut for Shader files to quickly create Shader Materials from them.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

<img width="609" alt="Screenshot 2022-12-31 at 18 13 35" src="https://user-images.githubusercontent.com/43449832/210150928-97666e53-e5eb-48ea-9e5b-bbc04c8d1ff3.png">

### If this enhancement will not be used often, can it be worked around with a few lines of script?

This can be worked around with an editor plugin maybe, but this is a very common operation and it would benefit everyone to be in core.

### Is there a reason why this should be core and not an add-on in the asset library?

This is about improving the usability out of the box.

**Comment by KoBeWi** [comment]
> Creating Shader Materials takes too many clicks. Assuming the Shader resource already exists, I need to create a new resource, select the Shader Material, etc.., and then I have to attach the Shader resource to it.

You don't. Just drag the shader to material slot and it will automatically create the material.

**Comment by realkotob** [comment]
@KoBeWi This is nice but this PR fixes usability for a different usecase.

The usecase we commonly do everytime is that we create a shader resource, and we want to create a material out of it to adjust the parameters before using it. 
It has almost never been the case where we create a Shader and we just want to attach it somewhere and start using it as-is.

Even if we did do it in the way you suggested, this would create a built-in resource, which is _never_ usually what is desired for material, as they are always intended to be checked into git and be reused throughout the project.

Again, the usability enhancement you suggested is nice, but it not helpful at all in the regular case when actually working on a game.

**Comment by KoBeWi** [comment]
> The usecase we commonly do everytime is that we create a shader resource, and we want to create a material file out of it to adjust the parameters before using it.

Is the shader shared between multiple materials? From my experience, it's more common to create material first and add a shader as built-in in the material.

**Comment by realkotob** [comment]
@KoBeWi Yes the Shader is shared between multiple Materials because it has a lot of public parameters that we modify in the Material resource. 

This is the most common workflow in all engines we used, in Unreal they call them Master Material instead of Shader and Material Instance instead of Shader Material. https://www.techarthub.com/the-anatomy-of-an-unreal-engine-master-material/

**Comment by jamesmintram** [comment]
Might be jumping ahead a little, but I have started a PR to implement this here: https://github.com/godotengine/godot/pull/70805

**Comment by jamesmintram** [comment]
Didn't realise @realkotob had already implemented this in https://github.com/godotengine/godot/pull/70776
